pkg/controller/render/render_controller.go

```diff
@@ -644,6 +645,10 @@ func generateRenderedMachineConfig(pool *mcfgv1.MachineConfigPool, configs []*mc
 	// the operator shouldn't stop the rest of the upgrade from progressing/completing.
 	if merged.Spec.OSImageURL != ctrlcommon.GetDefaultBaseImageContainer(&cconfig.Spec) {
 		merged.Annotations[ctrlcommon.OSImageURLOverriddenKey] = "true"
+		// Log a warning if the osImageURL is set using a tag instead of a digest
```

Comment:
thought (non-blocking): There are functions and types within `containers/image` that can be used for this purpose which can also handle edge-cases. Check out https://github.com/containers/image/tree/main/docker/reference. This module is already vendored into the MCO repo as well :smile: